### PR TITLE
Add tests for material_state_mouse_cursor.0.dart API example.

### PR DESCRIPTION
**Pull request by ksokolovskyi**
This PR contributes to https://github.com/flutter/flutter/issues/130459

### Description
- Updates `examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart` to allow enable/disable `ListTile` in tests;
- Adds tests for `examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart`.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

**Review by Renzo-Olivares — APPROVED** [review]
LGTM with small nits, thanks for adding this test!

File: examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          home: Scaffold(
+            body: example.MaterialStateMouseCursorExample(enabled: false),
+          ),
+        ),
+      );
+
+      final TestGesture gesture = await tester.createGesture(
+        kind: PointerDeviceKind.mouse,
+      );
+      await gesture.addPointer(
```
Comment (Renzo-Olivares) [inline examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart:65]: nit:  `addTearDown(gesture.removePointer);`

File: examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const MaterialApp(
+          home: Scaffold(
+            body: example.MaterialStateMouseCursorExample(enabled: true),
+          ),
+        ),
+      );
+
+      final TestGesture gesture = await tester.createGesture(
```
Comment (Renzo-Olivares) [inline examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart:37]: nit: `addTearDown(gesture.removePointer);`

File: examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart (hunk 1 above)
Comment (ksokolovskyi) [reply examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart:66]: thanks, added in the latest commit

File: examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart (hunk 2 above)
Comment (ksokolovskyi) [reply examples/api/test/material/material_state/material_state_mouse_cursor.0_test.dart:37]: thanks, added it in the latest commit

**Review by Renzo-Olivares — APPROVED** [review]
LGTM, thank you for your contribution!

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares thanks for the review!

File: examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart [hunk 3]
```diff
@@ -17,7 +17,10 @@ class MaterialStateMouseCursorExampleApp extends StatelessWidget {
       home: Scaffold(
         appBar: AppBar(title: const Text('MaterialStateMouseCursor Sample')),
         body: const Center(
-          child: MaterialStateMouseCursorExample(),
+          child: MaterialStateMouseCursorExample(
+            // Switch to get a different mouse cursor while hovering ListTile.
```
Comment (gspencergoog) [inline examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart:21]:
```suggestion
            // TRY THIS: Switch to get a different mouse cursor while hovering ListTile.
```

(We use this form in other examples)

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart (hunk 3 above)
Comment (ksokolovskyi) [reply examples/api/lib/material/material_state/material_state_mouse_cursor.0.dart]: @gspencergoog thanks for the suggestions!

**Comment by ksokolovskyi** [comment]
@Renzo-Olivares @gspencergoog thanks for the review!

**Pull request merged**